File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/java/io/github/lishangbu/avalon/dataset/entity/MoveTarget.java
package io.github.lishangbu.avalon.dataset.entity;

import io.github.lishangbu.avalon.hibernate.Flex;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.io.Serial;
import java.io.Serializable;
import lombok.Data;

/// 招式指向目标(MoveTarget)实体类
///
/// 表示招式指向目标的元数据，包括内部名称、显示名称与描述
///
/// @author lishangbu
/// @since 2025/08/20
@Data
@Entity
@Table(comment = "招式指向目标")
public class MoveTarget implements Serializable {
    @Serial private static final long serialVersionUID = 1L;

    /// 主键
    @Id
    @Flex
    @Column(comment = "主键")
    private Long id;

    /// 招式指向目标内部名称
    @Column(comment = "招式指向目标内部名称", length = 100)
    private String internalName;

    /// 招式指向目标名称
    @Column(comment = "招式指向目标名称", length = 100)
    private String name;

    /// 招式指向目标描述
    @Column(comment = "招式指向目标描述", length = 300)
    private String description;
}

File: modules/catalog/src/main/kotlin/io/github/lishangbu/avalon/catalog/domain/MoveTarget.kt
package io.github.lishangbu.avalon.catalog.domain


/**
 * 招式目标定义的领域视图。
 *
 * 这类数据用于表达一个招式可以指向谁，属于 Catalog 中的稳定参考数据。
 *
 * @property id 招式目标标识。
 * @property code 招式目标业务编码，供上下文和管理端引用。
 * @property name 招式目标展示名称。
 * @property description 招式目标说明，可为空。
 * @property sortingOrder 列表展示顺序。
 * @property enabled 当前招式目标是否启用。
 * @property version 乐观锁版本号。
 */
data class MoveTarget(
    val id: MoveTargetId,
    val code: String,
    val name: String,
    val description: String?,
    val sortingOrder: Int,
    val enabled: Boolean,
    val version: Long,
)

/**
 * 招式目标的轻量摘要。
 *
 * @property id 招式目标标识。
 * @property code 招式目标业务编码。
 * @property name 招式目标展示名称。
 */
data class MoveTargetSummary(
    val id: MoveTargetId,
    val code: String,
    val name: String,
)

/**
 * 创建或更新招式目标时使用的输入草稿。
 *
 * @property code 招式目标业务编码。
 * @property name 招式目标展示名称。
 * @property description 招式目标说明，可为空。
 * @property sortingOrder 列表展示顺序。
 * @property enabled 当前招式目标是否启用。
 */
data class MoveTargetDraft(
    val code: String,
    val name: String,
    val description: String?,
    val sortingOrder: Int,
    val enabled: Boolean,
)